packages/astro/src/vite-plugins/astro-swap-functions.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * A plugin for importing Astro's private 'transitions/swap-functions.js' module.
+ * This is temporary solution and can break at any time when end-user's update their `astro` version.
+ */
+import { readFile } from 'node:fs/promises';
+import { createRequire } from 'node:module';
+import { resolve } from 'node:path';
+import type { VitePlugin } from '../types.js';
+
+const require = createRequire(import.meta.url);
+const astroDist = resolve(require.resolve('astro/package.json'), '..');
+const swapFunctionEntry = resolve(astroDist, 'dist/transitions/swap-functions.js');
+
+const virtualModuleId = 'tutorialkit:astro-swap-functions';
+const resolvedId = `\0${virtualModuleId}`;
+
+export function astroSwapFunctions(): VitePlugin {
+  return {
+    name: 'tutorialkit-astro-swap-functions-plugin',
+    resolveId(id) {
+      if (id === virtualModuleId) {
+        return resolvedId;
+      }
+
+      return undefined;
+    },
+
+    async load(id) {
+      if (id === resolvedId) {
+        return await readFile(swapFunctionEntry, 'utf8');
```

Comment:
Wouldn't it be better to return an import to that path? If the module import other files, those import will fail if we return its content like this.

Reply:
I haven't heard of this problem before. How would that look in practice here? 

Is this error mentioned somewhere here https://rollupjs.org/plugin-development/#load?

Reply:
Ah no worries! I'll clarify. 

Consider relative imports. There is no way they can work as the module path is lost when just returning the module content as a virtual module. If you look at what we did for the `theme.css` file, we import it (instead of including its content) precisely to support local imports.

Reply:
Oh good point, I didn't realize this! 💯 

I'll change this to `export * from '${swapFunctionEntry}';`. There's no `default` export so maybe we don't need to handle that one here.